parsers/forth.c

```diff
@@ -0,0 +1,78 @@
+/*
+*   Copyright (c) 2023, Eric Forgeot
+*
+*   Based on work by Jon Strait
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your opinion) any later version.
+*
+*   This module contains functions for generating tags for Forth files
+*/
+
+/*
+*   INCLUDE FILES
+*/
+#include "general.h"	/* must always come first */
+
+#include <ctype.h>
+#include <string.h>
+
+#include "parse.h"
+#include "read.h"
+#include "vstring.h"
+#include "routines.h"
+#include "entry.h"
+
+/*
+*   DATA DEFINITIONS
+*/
+
+typedef enum {
+//   K_SECTION,
+   K_FUNCTION,
+} ForthKind;
+
+static kindDefinition ForthKinds[] = {
+	{ true, 's', "word", "words" },
```

Comment:
word?
It seems that this should be "function".
Am I wrong?

Reply:
In Forth there is no "function", the equivalent is called "word"

Reply:
In that case, `K_FUNCTION` is not a good name.
Should we use `s` as the letter for "word"? `w` looks better to me.

Reply:
I don't know how to label it "words", the result in geany is called "section", but maybe it should be changed into the geany's code instead

Reply:
> In that case, `K_FUNCTION` is not a good name. Should we use `s` as the letter for "word"? `w` looks better to me.

sure, why not. You can change it, but it looks like this code won't be used since you've made the forth.ctags in replacement? 

btw I've tested the generated optlib/forth.c into the geany code, and it didn't compile, I got a:
> parsers/forth.c:44:12: error: ‘parserDefinition’ {aka ‘struct sParserDefinition’} has no m
> ember named ‘versionCurrent’               
>   44 |         def->versionCurrent= 0;                                                   
>      |            ^~
> parsers/forth.c:45:12: error: ‘parserDefinition’ {aka ‘struct sParserDefinition’} has no m
> ember named ‘versionAge’                                                                     45 |         def->versionAge    = 0;

Reply:
Fixing the way to integrate ctags with Geany is the next step.
We should focus on making the forth parser work from CLI is the first.

Reply:
> Fixing the way to integrate ctags with Geany is the next step. We should focus on making the forth parser work from CLI is the first.

They've said they don't import the optlib from ctags, it doesn't look compatible I'm afraid:
https://github.com/geany/geany/pull/3377#issuecomment-1713642517